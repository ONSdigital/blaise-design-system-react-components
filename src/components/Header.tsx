import React, { ReactElement, ReactNode } from "react";

export interface NavigationLinks {
    id: string,
    label: string,
    endpoint: string
}

export interface Props {
    title: string
    signOutButton?: boolean
    noSave?: boolean
    signOutFunction?: () => void
    navigationLinks?: NavigationLinks[]
    currentLocation?: string
    createNavLink?: (id: string, label: string, endpoint: string) => ReactNode
}

function Header({
    title, signOutButton, noSave, signOutFunction, navigationLinks, currentLocation, createNavLink,
}: Props): ReactElement {
    const createLink = (id: string, label: string, endpoint: string) => {
        if (createNavLink) {
            return createNavLink(id, label, endpoint);
        }
        return (
            <a className="ons-navigation__link" id={id} href={endpoint} role="link">
                {label}
            </a>
        );
    };

    let signOutText = "Save and sign out";
    if (noSave) {
        signOutText = "Sign out";
    }
    return (
        <header className="ons-header ons-header--internal">
            <div className="ons-header__top" role="banner">
                <div className="ons-container">
                    <div
                        className="ons-header__grid-top ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap "
                    >
                        <div className="ons-grid__col ons-col-auto">
                            <a className="header__logo-link" href="/">
                                <picture>
                                    {/* <source media="(max-width: 499px)" srcSet="/img/ons-logo-stacked-neg-en.svg" */}
                                    {/*        alt="Office for National Statistics logo"/> */}
                                    <img
                                        className="ons-header__logo"
                                        src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                                        alt="Office for National Statistics logo"
                                    />
                                </picture>
                            </a>
                        </div>
                    </div>
                </div>
            </div>
            <div className="ons-header__main">
                <div className="ons-container">
                    <div
                        className="ons-grid ons-grid-flex ons-grid-flex--between ons-grid-flex--vertical-center ons-grid-flex--no-wrap"
                    >
                        <div className="ons-grid__col ons-col-auto ons-u-flex-shrink">
                            <div className="ons-header__title">{title}</div>
                        </div>
                        {
                            signOutButton
                            && (
                                <div className="ons-grid__col ons-col-auto ons-u-flex-no-shrink">
                                    <button
                                        id="signout-button"
                                        data-test-id="signout-button"
                                        className="ons-u-d-no@2xs@m ons-btn--ghost ons-btn--link ons-js-submit-btn"
                                        onClick={() => signOutFunction && signOutFunction()}
                                        type="button"
                                    >
                                        <span className="ons-btn__inner">
                                            <span className="ons-btn__text">{signOutText}</span>
                                            <svg
                                                className="ons-icon ons-u-ml-2xs"
                                                viewBox="0 0 12 12"
                                                xmlns="http://www.w3.org/2000/svg"
                                                focusable="false"
                                                fill="currentColor" 
                                                role="img" 
                                                aria-hidden="true"
                                            >
                                                <path
                                                    d="M13.85,7.65l-2.5-2.5a.5.5,0,0,0-.71,0,.48.48,0,0,0-.15.36V7h-3a.5.5,0,0,0-.5.5v1a.5.5,0,0,0,.5.5h3v1.5A.49.49,0,0,0,11,11a.48.48,0,0,0,.34-.14l2.51-2.5a.49.49,0,0,0,0-.68Z"
                                                    transform="translate(-2 -2)"
                                                />
                                                <path
                                                    d="M8.5,14h-6a.5.5,0,0,1-.5-.5V2.5A.5.5,0,0,1,2.5,2h6a.5.5,0,0,1,.5.5V3a.5.5,0,0,1-.5.5h-5v9h5A.5.5,0,0,1,9,13v.5A.5.5,0,0,1,8.5,14Z"
                                                    transform="translate(-2 -2)"
                                                />
                                            </svg>
                                        </span>
                                    </button>
                                </div>
                            )
                        }
                    </div>
                </div>
            </div>
            {
                (navigationLinks && navigationLinks.length !== 0)
                && (
                    <div className="ons-navigation-wrapper">
                        <div className="ons-container ons-container--gutterless@xxs@l">
                            <nav
                                className="ons-navigation ons-navigation--main ons-js-navigation"
                                id="main-nav"
                                aria-label="Main menu"
                                data-analytics="header-navigation"
                                role="navigation"
                            >
                                <ul className="ons-navigation__list">
                                    {
                                        navigationLinks.map(({ id, label, endpoint }, index) => (
                                            <li
                                                key={index}
                                                className={`ons-navigation__item  ${(currentLocation === endpoint ? "ons-navigation__item--active" : "")}`}
                                            >
                                                {createLink(id, label, endpoint)}
                                            </li>
                                        ))
                                    }
                                </ul>
                            </nav>
                        </div>
                    </div>
                )
            }
        </header>
    );
}

export default Header;
