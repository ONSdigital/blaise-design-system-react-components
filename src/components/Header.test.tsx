import React from "react";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Header, { Props } from "./Header";

const testProps: Props = {
    title: "ONS Service",
    navigationLinks: [
        { id: "home-link", label: "Home", endpoint: "/" },
        { id: "deploy-questionnaire-link", label: "Deploy a questionnaire", endpoint: "/deploy" },
        { id: "audit-logs-link", label: "View deployment history", endpoint: "/history" },
        { id: "blaise-status-link", label: "Check Blaise status", endpoint: "/status" },
    ],
};

describe.only("Check default Header:", () => {
    it("matches Snapshot", () => {
        const wrapper = render(
            <Header title={testProps.title} />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should render with the title displayed", () => {
        render(
            <Header title={testProps.title} />,
        );
        expect(screen.getByText(testProps.title)).toBeVisible();
    });

    it("should not show the signout button by default", () => {
        render(
            <Header title={testProps.title} />,
        );
        expect(screen.queryAllByText(/Save and sign out/)).toStrictEqual([]);
        expect(screen.queryAllByText(/Sign out/)).toStrictEqual([]);
    });

    it("should not show the navigation by default", () => {
        render(
            <Header title={testProps.title} />,
        );
        expect(screen.queryByRole(/navigation/)).toStrictEqual(null);
    });
});

describe("Check Header with sign out button:", () => {
    it("matches Snapshot with sign out button", () => {
        const wrapper = render(
            <Header title={testProps.title} signOutButton signOutFunction={jest.fn()} />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("shows sign out button", () => {
        render(
            <Header title={testProps.title} signOutButton signOutFunction={jest.fn()} />,
        );
        expect(screen.getByText(/Save and sign out/)).toBeVisible();
    });

    it("shows sign out button with special text", () => {
        render(
            <Header title={testProps.title} signOutButton noSave signOutFunction={jest.fn()} />,
        );
        expect(screen.getByText(/Sign out/)).toBeDefined();
        expect(screen.queryAllByText(/Save and sign out/)).toStrictEqual([]);
    });

    it("passes in the sign out function correctly button", async () => {
        const mockFunction = jest.fn();

        render(
            <Header title={testProps.title} signOutButton signOutFunction={mockFunction} />,
        );
        const signOutButton = screen.getByText(/Save and sign out/);
        await userEvent.click(signOutButton);

        expect(mockFunction).toHaveBeenCalledTimes(1);
    });
});

describe("Check Header with navigation bar:", () => {
    it("matches Snapshot with navigation", () => {
        const wrapper = render(
            <Header title={testProps.title} navigationLinks={testProps.navigationLinks} currentLocation="/" />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("shows the navigation with links by name", () => {
        render(
            <Header title={testProps.title} navigationLinks={testProps.navigationLinks} currentLocation="/" />,
        );
        expect(screen.queryByRole(/navigation/)).not.toStrictEqual(null);
        expect(screen.getByRole("link", { name: /Home/ })).toBeVisible();
        expect(screen.getByRole("link", { name: /Deploy a questionnaire/ })).toBeVisible();
        expect(screen.getByRole("link", { name: /View deployment history/ })).toBeVisible();
        expect(screen.getByRole("link", { name: /Check Blaise status/ })).toBeVisible();
    });

    it("has navigation links with IDs for integrated tests)", () => {
        render(
            <Header title={testProps.title} navigationLinks={testProps.navigationLinks} currentLocation="/" />,
        );
        expect(screen.queryByRole(/navigation/)).not.toStrictEqual(null);
        expect(screen.getByRole("link", { name: /Home/ })).toHaveAttribute("id", "home-link");
        expect(screen.getByRole("link", { name: /Deploy a questionnaire/ })).toHaveAttribute("id", "deploy-questionnaire-link");
        expect(screen.getByRole("link", { name: /View deployment history/ })).toHaveAttribute("id", "audit-logs-link");
        expect(screen.getByRole("link", { name: /Check Blaise status/ })).toHaveAttribute("id", "blaise-status-link");
    });

    it("shows active link", () => {
        render(
            <Header title={testProps.title} navigationLinks={testProps.navigationLinks} currentLocation="/deploy" />,
        );
        expect(screen.getByRole("link", { name: /Deploy a questionnaire/ }).parentElement).toHaveClass("ons-navigation__item--active");
    });
});
