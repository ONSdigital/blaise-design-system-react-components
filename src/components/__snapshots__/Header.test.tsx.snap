// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Header with navigation bar: matches Snapshot with navigation 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="ons-header ons-header--internal"
      >
        <div
          class="ons-header__top"
          role="banner"
        >
          <div
            class="ons-container"
          >
            <div
              class="ons-header__grid-top ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap "
            >
              <div
                class="ons-grid__col ons-col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="ons-header__logo"
                      src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ons-header__main"
        >
          <div
            class="ons-container"
          >
            <div
              class="ons-grid ons-grid-flex ons-grid-flex--between ons-grid-flex--vertical-center ons-grid-flex--no-wrap"
            >
              <div
                class="ons-grid__col ons-col-auto ons-u-flex-shrink"
              >
                <div
                  class="ons-header__title"
                >
                  ONS Service
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ons-navigation-wrapper"
        >
          <div
            class="ons-container ons-container--gutterless@xxs@l"
          >
            <nav
              aria-label="Main menu"
              class="ons-navigation ons-navigation--main ons-js-navigation"
              data-analytics="header-navigation"
              id="main-nav"
              role="navigation"
            >
              <ul
                class="ons-navigation__list"
              >
                <li
                  class="ons-navigation__item  ons-navigation__item--active"
                >
                  <a
                    class="ons-navigation__link"
                    href="/"
                    id="home-link"
                    role="link"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="ons-navigation__item  "
                >
                  <a
                    class="ons-navigation__link"
                    href="/deploy"
                    id="deploy-questionnaire-link"
                    role="link"
                  >
                    Deploy a questionnaire
                  </a>
                </li>
                <li
                  class="ons-navigation__item  "
                >
                  <a
                    class="ons-navigation__link"
                    href="/history"
                    id="audit-logs-link"
                    role="link"
                  >
                    View deployment history
                  </a>
                </li>
                <li
                  class="ons-navigation__item  "
                >
                  <a
                    class="ons-navigation__link"
                    href="/status"
                    id="blaise-status-link"
                    role="link"
                  >
                    Check Blaise status
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="ons-header ons-header--internal"
    >
      <div
        class="ons-header__top"
        role="banner"
      >
        <div
          class="ons-container"
        >
          <div
            class="ons-header__grid-top ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap "
          >
            <div
              class="ons-grid__col ons-col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="ons-header__logo"
                    src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ons-header__main"
      >
        <div
          class="ons-container"
        >
          <div
            class="ons-grid ons-grid-flex ons-grid-flex--between ons-grid-flex--vertical-center ons-grid-flex--no-wrap"
          >
            <div
              class="ons-grid__col ons-col-auto ons-u-flex-shrink"
            >
              <div
                class="ons-header__title"
              >
                ONS Service
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ons-navigation-wrapper"
      >
        <div
          class="ons-container ons-container--gutterless@xxs@l"
        >
          <nav
            aria-label="Main menu"
            class="ons-navigation ons-navigation--main ons-js-navigation"
            data-analytics="header-navigation"
            id="main-nav"
            role="navigation"
          >
            <ul
              class="ons-navigation__list"
            >
              <li
                class="ons-navigation__item  ons-navigation__item--active"
              >
                <a
                  class="ons-navigation__link"
                  href="/"
                  id="home-link"
                  role="link"
                >
                  Home
                </a>
              </li>
              <li
                class="ons-navigation__item  "
              >
                <a
                  class="ons-navigation__link"
                  href="/deploy"
                  id="deploy-questionnaire-link"
                  role="link"
                >
                  Deploy a questionnaire
                </a>
              </li>
              <li
                class="ons-navigation__item  "
              >
                <a
                  class="ons-navigation__link"
                  href="/history"
                  id="audit-logs-link"
                  role="link"
                >
                  View deployment history
                </a>
              </li>
              <li
                class="ons-navigation__item  "
              >
                <a
                  class="ons-navigation__link"
                  href="/status"
                  id="blaise-status-link"
                  role="link"
                >
                  Check Blaise status
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Check Header with sign out button: matches Snapshot with sign out button 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="ons-header ons-header--internal"
      >
        <div
          class="ons-header__top"
          role="banner"
        >
          <div
            class="ons-container"
          >
            <div
              class="ons-header__grid-top ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap "
            >
              <div
                class="ons-grid__col ons-col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="ons-header__logo"
                      src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ons-header__main"
        >
          <div
            class="ons-container"
          >
            <div
              class="ons-grid ons-grid-flex ons-grid-flex--between ons-grid-flex--vertical-center ons-grid-flex--no-wrap"
            >
              <div
                class="ons-grid__col ons-col-auto ons-u-flex-shrink"
              >
                <div
                  class="ons-header__title"
                >
                  ONS Service
                </div>
              </div>
              <div
                class="ons-grid__col ons-col-auto ons-u-flex-no-shrink"
              >
                <button
                  class="ons-u-d-no@2xs@m ons-btn--ghost ons-btn--link ons-js-submit-btn"
                  data-test-id="signout-button"
                  id="signout-button"
                  type="button"
                >
                  <span
                    class="ons-btn__inner"
                  >
                    <span
                      class="ons-btn__text"
                    >
                      Save and sign out
                    </span>
                    <svg
                      aria-hidden="true"
                      class="ons-icon ons-u-ml-2xs"
                      fill="currentColor"
                      focusable="false"
                      role="img"
                      viewBox="0 0 12 12"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.85,7.65l-2.5-2.5a.5.5,0,0,0-.71,0,.48.48,0,0,0-.15.36V7h-3a.5.5,0,0,0-.5.5v1a.5.5,0,0,0,.5.5h3v1.5A.49.49,0,0,0,11,11a.48.48,0,0,0,.34-.14l2.51-2.5a.49.49,0,0,0,0-.68Z"
                        transform="translate(-2 -2)"
                      />
                      <path
                        d="M8.5,14h-6a.5.5,0,0,1-.5-.5V2.5A.5.5,0,0,1,2.5,2h6a.5.5,0,0,1,.5.5V3a.5.5,0,0,1-.5.5h-5v9h5A.5.5,0,0,1,9,13v.5A.5.5,0,0,1,8.5,14Z"
                        transform="translate(-2 -2)"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="ons-header ons-header--internal"
    >
      <div
        class="ons-header__top"
        role="banner"
      >
        <div
          class="ons-container"
        >
          <div
            class="ons-header__grid-top ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap "
          >
            <div
              class="ons-grid__col ons-col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="ons-header__logo"
                    src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ons-header__main"
      >
        <div
          class="ons-container"
        >
          <div
            class="ons-grid ons-grid-flex ons-grid-flex--between ons-grid-flex--vertical-center ons-grid-flex--no-wrap"
          >
            <div
              class="ons-grid__col ons-col-auto ons-u-flex-shrink"
            >
              <div
                class="ons-header__title"
              >
                ONS Service
              </div>
            </div>
            <div
              class="ons-grid__col ons-col-auto ons-u-flex-no-shrink"
            >
              <button
                class="ons-u-d-no@2xs@m ons-btn--ghost ons-btn--link ons-js-submit-btn"
                data-test-id="signout-button"
                id="signout-button"
                type="button"
              >
                <span
                  class="ons-btn__inner"
                >
                  <span
                    class="ons-btn__text"
                  >
                    Save and sign out
                  </span>
                  <svg
                    aria-hidden="true"
                    class="ons-icon ons-u-ml-2xs"
                    fill="currentColor"
                    focusable="false"
                    role="img"
                    viewBox="0 0 12 12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.85,7.65l-2.5-2.5a.5.5,0,0,0-.71,0,.48.48,0,0,0-.15.36V7h-3a.5.5,0,0,0-.5.5v1a.5.5,0,0,0,.5.5h3v1.5A.49.49,0,0,0,11,11a.48.48,0,0,0,.34-.14l2.51-2.5a.49.49,0,0,0,0-.68Z"
                      transform="translate(-2 -2)"
                    />
                    <path
                      d="M8.5,14h-6a.5.5,0,0,1-.5-.5V2.5A.5.5,0,0,1,2.5,2h6a.5.5,0,0,1,.5.5V3a.5.5,0,0,1-.5.5h-5v9h5A.5.5,0,0,1,9,13v.5A.5.5,0,0,1,8.5,14Z"
                      transform="translate(-2 -2)"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Check default Header: matches Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="ons-header ons-header--internal"
      >
        <div
          class="ons-header__top"
          role="banner"
        >
          <div
            class="ons-container"
          >
            <div
              class="ons-header__grid-top ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap "
            >
              <div
                class="ons-grid__col ons-col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="ons-header__logo"
                      src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ons-header__main"
        >
          <div
            class="ons-container"
          >
            <div
              class="ons-grid ons-grid-flex ons-grid-flex--between ons-grid-flex--vertical-center ons-grid-flex--no-wrap"
            >
              <div
                class="ons-grid__col ons-col-auto ons-u-flex-shrink"
              >
                <div
                  class="ons-header__title"
                >
                  ONS Service
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="ons-header ons-header--internal"
    >
      <div
        class="ons-header__top"
        role="banner"
      >
        <div
          class="ons-container"
        >
          <div
            class="ons-header__grid-top ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap "
          >
            <div
              class="ons-grid__col ons-col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="ons-header__logo"
                    src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ons-header__main"
      >
        <div
          class="ons-container"
        >
          <div
            class="ons-grid ons-grid-flex ons-grid-flex--between ons-grid-flex--vertical-center ons-grid-flex--no-wrap"
          >
            <div
              class="ons-grid__col ons-col-auto ons-u-flex-shrink"
            >
              <div
                class="ons-header__title"
              >
                ONS Service
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
